feat(dashboards): Allow project column in Issue Widgets

edwardgou-sentry opened the pull request:
Allows selection of project column for Issue Widgets
Refactors first and last seen to use default date renderers instead

![image](https://user-images.githubusercontent.com/83961295/152571037-de1f7c1f-fbce-4c0b-96df-435bad307aa4.png)

edwardgou-sentry committed:
adds project to issue field renderer

refactors first and last seen to use default date renderers

edwardgou-sentry committed:
fixed type

priscilawebdev commented on static/app/views/dashboardsV2/widgetCard/issueWidgetQueries.tsx:
```diff
@@ -124,8 +124,17 @@ class IssueWidgetQueries extends React.Component<Props, State> {
     tableResults.forEach(group => {
-      const {id, shortId, title, lifetime, filtered, count, userCount, ...resultProps} =
-        group;
+      const {
+        id,
+        shortId,
+        title,
+        lifetime,
+        filtered,
+        count,
+        userCount,
+        project,
+        ...resultProps
+      } = group;
```
```suggestion
```

priscilawebdev commented on static/app/views/dashboardsV2/widgetCard/issueWidgetQueries.tsx:
```diff
@@ -124,8 +124,17 @@ class IssueWidgetQueries extends React.Component<Props, State> {
     GroupStore.add(tableResults);
     const transformedTableResults: TableDataRow[] = [];
     tableResults.forEach(group => {
```
```suggestion
    tableResults.forEach({
        id,
        shortId,
        title,
        lifetime,
        filtered,
        count,
        userCount,
        project,
        ...resultProps
      }) => {
```

priscilawebdev commented:
since we are working on these files,  can we replace `group.project.id` with `project.id`?

https://github.com/getsentry/sentry/blob/a1c2243b2660b3843f3f26d414afe668c94f287a/static/app/views/dashboardsV2/widgetCard/issueWidgetQueries.tsx#L187

priscilawebdev commented on static/app/views/dashboardsV2/widgetCard/issueWidgetQueries.tsx:
```diff
@@ -141,6 +150,7 @@ class IssueWidgetQueries extends React.Component<Props, State> {
         'issue.id': id,
         issue: shortId,
         title,
+        project: project?.slug,
```
can project be `undefined/null `here? if yes, should we update the types?

edwardgou-sentry replied:
project should be defined in Group as far as I can tell. I haven't seen a case where it hasn't been defined yet. I've updated this to just `project.slug` now. Thanks!

edwardgou-sentry committed:
did feedback

moved destructuring into function args
removed optional from project
fixed test case

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/issue-widget-project-field

edwardgou-sentry committed:
missed removing firstSeen during merge

priscilawebdev reviewed (approved):
code lgtm
